src/videojs-hls.js

```diff
@@ -335,13 +337,43 @@ videojs.Hls.prototype.addCuesForMetadata_ = function(segmentInfo) {
     for (i = 0; i < metadata.frames.length; i++) {
       frame = metadata.frames[i];
       time = segmentOffset + ((metadata.pts - minPts) * 0.001);
-      cue = new window.VTTCue(time, time, frame.value || frame.url || '');
+      cue = new Cue(time, time, frame.value || frame.url || '');
       cue.frame = frame;
+      cue.value = frame;
       cue.pts_ = metadata.pts;
       textTrack.addCue(cue);
     }
     segmentInfo.pendingMetadata.shift();
   }
+
+  if (cue && cue.frame) {
```

Comment:
It appears this code will only run for the last cue created in the `for` loop above.